src/components/QuizQuestions/index.tsx

```diff
@@ -10,11 +10,23 @@ export default function QuizQuestions() {
   const nextQuestion = state?.quiz.currentQuestion?.next_question;
   const questionTypes = getQuestionTypes(nextQuestion?.type);
 
-  return (
-    <>
-      {questionTypes.isOpenQuestion && <OpenTextQuestion key={nextQuestion?.id} />}
-      {questionTypes.isCoverQuestion && <CoverTypeQuestion key={nextQuestion?.id} />}
-      {questionTypes.isSelectQuestion && <SelectTypeQuestion key={nextQuestion?.id} />}
-    </>
-  );
+  if (nextQuestion) {
+    return (
+      <>
+        {questionTypes.isOpenQuestion && <OpenTextQuestion key={nextQuestion?.id} />}
+        {questionTypes.isCoverQuestion && (
```

Comment:
Why does CoverQuestion take a bunch of props but others rely on context?

Reply:
Just refactored one component so the PR is not big and also to make sure we agree on those changes before changing all of the components